src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_formula_generator.py

```diff
@@ -100,6 +101,38 @@ def _get_builder(
         )
         return builder
 
+    def _get_graph_and_grid_component_successors(
+        self,
+    ) -> tuple[ComponentGraph, set[component.Component]]:
```

Comment:
Getting the component graph is only one line of code, wouldn't it make more sense to have this function instead?

```suggestion
    def _get_grid_component_successors(
        self,
        component_graph: ComponentGraph,
    ) -> set[component.Component]:
```

I find the name and return type too convoluted otherwise.

Reply:
Not sure why didn't you take the component graph as an argument, you could also make it optional and get it inside the function if `None` was provided. But I guess it is OK like this too, at the end it shouldn't be expensive to get the component graph anyway.